mempool/mempool.go

```diff
@@ -142,12 +142,10 @@ func (mem *Mempool) SetLogger(l log.Logger) {
 	mem.logger = l
 }
 
-// SetFilter sets a filter for mempool to only accept txs for which f(tx)
-// returns true.
-func (mem *Mempool) SetFilter(f func(types.Tx) bool) {
-	mem.proxyMtx.Lock()
-	mem.filter = f
-	mem.proxyMtx.Unlock()
+// Filter sets a filter for mempool to only accept txs for which f(tx) returns
+// true.
+func Filter(f func(types.Tx) bool) MempoolOption {
```

Comment:
(Sorry for the bikeshed) Can this still be named `SetFilter`? I feel like that was clearer, since this isn't the actual filter.

Reply:
WithFilter sounds better to me